### PR TITLE
feat: support global name

### PR DESCRIPTION
**Pull request by Lulalaby**

### CODE REVIEW

**Comment by Lulalaby** [comment]
I'm not sure if that's entirely correct.
From what I see y'all reference the `next` version which supports global_name.
We've noticed the missing functionality while using the bot.
I'm kinda a beginner on rust, so sorry if there's anything wrong ❤️

**Comment by GnomedDev** [comment]
Hey, thanks for the PR! Can you refactor this to use [`Option::or`](https://doc.rust-lang.org/std/option/enum.Option.html#method.or)? That would be perfect, then I'll merge and that will be good.

File: src/funcs.rs [hunk 1]
```diff
@@ -465,7 +465,7 @@ pub fn clean_msg(
             .unwrap();
         }
 
-        let said_name = nickname.unwrap_or_else(|| member_nick.unwrap_or(&user.name));
+        let said_name = nickname.unwrap_or_else(|| member_nick.unwrap_or_else(|| &user.global_name.unwrap_or(&user.name)));
```
Comment (Lulalaby) [inline src/funcs.rs:468]:
```suggestion
        let said_name = nickname.unwrap_or_else(|| member_nick.unwrap_or_else(|| &user.global_name.or(&user.name)));
```

you mean like that? sorry i'm not familiar enough

Comment (GnomedDev) [reply src/funcs.rs:468]: Yeah, but for the other two unwrap_or_else calls aswell

**Review by Lulalaby — COMMENTED** [review]
Should work.. I guess?

**Pull request merged**